lib/matplotlib/tests/test_font_manager.py

```diff
@@ -31,6 +31,20 @@ def test_font_priority():
     assert cmap[8729] == 30
 
 
+def test_ttflist_weight():
+    assert all(isinstance(f.weight, six.string_types)
+               for f in fontManager.ttflist)
+
+
+def test_score_weight():
+    assert (0 ==
+            fontManager.score_weight("regular", "regular") ==
+            fontManager.score_weight("bold", "bold") <
+            fontManager.score_weight("normal", "regular") ==
+            fontManager.score_weight(400, 400) <
```

Comment:
The numerical weight of a font is inferred from the weight name of the font and thus is inaccurate.
For example, FiraSans-Hair, which is very thin, and FiraSans-Regular has the same numeric weight, 400, and thus they have the same priority for the previous `score_weight` implementation.